h2o-algos/src/main/java/hex/tree/dt/mrtasks/GetClassCountsMRTask.java

```diff
@@ -0,0 +1,67 @@
+package hex.tree.dt.mrtasks;
+
+import org.apache.commons.math3.util.Precision;
+import water.MRTask;
+import water.fvec.Chunk;
+
+/**
+ * MR task for counting classes.
+ */
+public class GetClassCountsMRTask extends MRTask<GetClassCountsMRTask> {
+    public int _numClasses;
+    // counts of samples for each class, class corresponds to the index in array: [count0, count1, ...]
+    public int[] _countsByClass;
+
+    private final double[][] _featuresLimits;
+
+    int LIMIT_MIN = 0;
+    int LIMIT_MAX = 1;
+
+    public GetClassCountsMRTask(double[][] featuresLimits, int numClasses) {
+        _numClasses = numClasses;
+        _countsByClass = new int[_numClasses];
+        _featuresLimits = featuresLimits;
+    }
+
+    @Override
+    public void map(Chunk[] cs) {
+        // deep copy of countsByClass array so the reduce phase performs correctly
+        {
+            int[] tmpCounts = new int[_numClasses];
+            for (int c = 0; c < _numClasses; c++) {
+                tmpCounts[c] = 0;
+            }
+            _countsByClass = tmpCounts;
+        }
+        int classColumn = cs.length - 1; // the last column
+        int numRows = cs[0]._len;
+        boolean conditionsFailed;
+        // select only rows that fulfill all conditions
+        for (int row = 0; row < numRows; row++) {
+            conditionsFailed = false;
+            // - 1 because of the class column - don't check limits on it
+            for (int column = 0; column < cs.length - 1 /*exclude prediction column*/; column++) {
```

Comment:
replace 

for (int column=0; column<cs.length-1; column++)

with

for (int column=0; column < classColumn; column++)

This could give us some speedup.